### PR TITLE
Reno 3904/update newsletter signup component

### PR DESCRIPTION
**Pull request by isastettler**
Fixes [JIRA ticket](https://newyorkpubliclibrary.atlassian.net/browse/RENO-3904)

## This PR does the following:

- Replaces  `newsletterSignupType` with `highlightColor` prop
- Updates theme file to use highlightColor
- Adds `showDefaultPrivacyLink` prop to account for the DXP use case
- Removes default values for Headings and description texts
- Updates all Heading props to be required
- Update `title` prop to be rendered as a "heading3" size when passed a string
- Updates storybook accordingly
- Removes unnecessary snapshot tests
- Updates tests

## How has this been tested?

Tested in storybook and through existing unit tests.

## Accessibility concerns or updates

- N/A

### Checklist:

- [x] I have updated the Storybook documentation accordingly.
- [ ] I have added relevant accessibility documentation for this pull request.
- [x] All new and existing tests passed.

### Front End Review:

- [ ] Review the [Vercel preview](https://nypl-design-system-git-reno-3904-update-newsletter-7afc80-nypl.vercel.app/?path=/docs/components-form-elements-newslettersignup--docs) deployment once it is ready.

### CODE REVIEW

File: src/components/NewsletterSignup/NewsletterSignup.mdx
Comment (EdwinGuzman) [inline src/components/NewsletterSignup/NewsletterSignup.mdx]: Update the latest to `Prerelease`. When it's about to get released, we'll update the version.

File: src/components/NewsletterSignup/NewsletterSignup.tsx
Comment (EdwinGuzman) [inline src/components/NewsletterSignup/NewsletterSignup.tsx]: Typo: "Privaly"

File: src/components/NewsletterSignup/NewsletterSignup.tsx
Comment (EdwinGuzman) [inline src/components/NewsletterSignup/NewsletterSignup.tsx]: You can create an array with these values and then create a type based on the array. Then you can reuse the array in the .stories.tsx file.

```
export const highlightTypesArray = [...];
export type HighlightTypes = typeof highlightTypesArray[number];
```

Check `Banner.tsx` for an example.

File: src/components/NewsletterSignup/NewsletterSignup.stories.tsx
Comment (EdwinGuzman) [inline src/components/NewsletterSignup/NewsletterSignup.stories.tsx]: If you create the array with values in the main file, you can import it and use it here as `options: highlightTypesArray,`. This way, it's not hard coded and if you add a new color in the array, this will automatically get updated.

File: src/components/NewsletterSignup/NewsletterSignup.stories.tsx
Comment (EdwinGuzman) [inline src/components/NewsletterSignup/NewsletterSignup.stories.tsx]: Nit: you can use the shorthand
```
confirmationHeading,
confirmationText: undefined,
descriptionText,
... 
```

File: src/components/NewsletterSignup/NewsletterSignup.stories.tsx
Comment (EdwinGuzman) [inline src/components/NewsletterSignup/NewsletterSignup.stories.tsx]: This is because the props can take in string or JSX.Element values? This makes sense and not for this PR but there are other components with similar prop value issues. Something like this would be nice in a util file and function so other Stories can reuse this (but again, you don't need to do this for this PR).

Normally, for these type of cases we turn off the control for the prop and give explicit values of all possible input values in the docs page. Just curious, when I do pass in HTML values for these props, it doesn't seem to work. What have you tried to pass that works?

File: src/components/NewsletterSignup/NewsletterSignup.stories.tsx
Comment (EdwinGuzman) [inline src/components/NewsletterSignup/NewsletterSignup.stories.tsx:212]: Related to the comment above. The next example has the h4 element for `title` and this one has  a string 👍

File: src/components/NewsletterSignup/NewsletterSignup.tsx
Comment (EdwinGuzman) [inline src/components/NewsletterSignup/NewsletterSignup.tsx]: I think you can rename this `showPrivacyLink`

File: src/components/NewsletterSignup/NewsletterSignup.tsx
Comment (EdwinGuzman) [inline src/components/NewsletterSignup/NewsletterSignup.tsx:152]: nitpick: are you using the class names? If not, can you remove them?

File: src/components/NewsletterSignup/NewsletterSignup.tsx
Comment (EdwinGuzman) [inline src/components/NewsletterSignup/NewsletterSignup.tsx]: This is not a change you made but I just noticed that `alignItems="flex-start"` is already in `styles.pitch` so it can be removed here.

File: src/theme/components/newsletterSignup.ts
Comment (EdwinGuzman) [inline src/theme/components/newsletterSignup.ts:55]: This is no longer used. Can you remove this?

**Review by EdwinGuzman — CHANGES_REQUESTED** [review]
Requesting changes but overall the update is great.

For the Storybook doc page, can you give the examples various random `highlightColor` values? Just to show the various colors that are available for the component.

File: src/theme/components/newsletterSignup.ts
Comment (isastettler) [reply src/theme/components/newsletterSignup.ts:55]: This is used to target an element rendered by the Form component. The id passed to the `<Form>` element gets `-parent` attached and the styling is modifying the rendered `<div>` with `id-"newsletter-form-parent"`.

File: src/components/NewsletterSignup/NewsletterSignup.mdx
Comment (isastettler) [reply src/components/NewsletterSignup/NewsletterSignup.mdx]: Done.

File: src/components/NewsletterSignup/NewsletterSignup.tsx
Comment (isastettler) [reply src/components/NewsletterSignup/NewsletterSignup.tsx]: Thanks for catching. Fixed.

File: src/components/NewsletterSignup/NewsletterSignup.stories.tsx
Comment (isastettler) [reply src/components/NewsletterSignup/NewsletterSignup.stories.tsx]: I should have seen this. Thanks. Fixed.

File: src/components/NewsletterSignup/NewsletterSignup.stories.tsx
Comment (isastettler) [reply src/components/NewsletterSignup/NewsletterSignup.stories.tsx]: Removed the code. Disabled the control for the title prop.

File: src/components/NewsletterSignup/NewsletterSignup.tsx
Comment (isastettler) [reply src/components/NewsletterSignup/NewsletterSignup.tsx]: Done.

File: src/components/NewsletterSignup/NewsletterSignup.tsx
Comment (isastettler) [reply src/components/NewsletterSignup/NewsletterSignup.tsx]: Done.

File: src/components/NewsletterSignup/NewsletterSignup.tsx
Comment (isastettler) [reply src/components/NewsletterSignup/NewsletterSignup.tsx]: Hey yes I know about that syntax. I chose the other way because I personally like better when the type name is shown in the storybook arguments table rather than the list: 
Showing all possible colors
vs
Only showing the type

Comment (isastettler) [reply src/components/NewsletterSignup/NewsletterSignup.tsx]: Updated it though :)

File: src/components/NewsletterSignup/NewsletterSignup.stories.tsx
Comment (isastettler) [reply src/components/NewsletterSignup/NewsletterSignup.stories.tsx]: Done.

File: src/components/NewsletterSignup/NewsletterSignup.mdx
Comment (EdwinGuzman) [inline src/components/NewsletterSignup/NewsletterSignup.mdx]: Sorry, I meant that this should be
```
| Latest     | `Prerelease` |
```

**Review by EdwinGuzman — COMMENTED** [review]
This looks good but the only comment left is adding information in the Docs page for the `highlightColor` prop and showing a few examples of the various values. Or update the existing examples on the page with a `highlightColor` value (but still do add a small section to describe the `highlightColor` prop and how it will update the color on the border.

File: src/components/NewsletterSignup/NewsletterSignup.tsx
Comment (EdwinGuzman) [reply src/components/NewsletterSignup/NewsletterSignup.tsx]: Yes, it is might nicer and clearer with all the values listed down :\

**Comment by EdwinGuzman** [comment]
Sorry, two more things:
- Update the changelog with this update.
- Update the local `newsletterSignupChangelogData.ts` file with this update. Here is an example: https://github.com/NYPL/nypl-design-system/pull/1614/files#diff-e094863f0831a7133246108e781c5d7696723384e8eb8f319319f90f99f3194e

**Review by EdwinGuzman — APPROVED** [review]
This looks good to me, thanks! Do wait on @bigfishdesign13 for a review as well.

It seems the updates to the component were somewhat minor and most of the work here is documentation. Do you need/want to test this in SCOUT? If so, we can create a release candidate npm version to test it there or in Turbine. Otherwise, I think it's good to go.

File: src/components/NewsletterSignup/NewsletterSignup.mdx
Comment (isastettler) [reply src/components/NewsletterSignup/NewsletterSignup.mdx]: LOL. Changed.

File: src/components/NewsletterSignup/NewsletterSignup.tsx
Comment (bigfishdesign13) [inline src/components/NewsletterSignup/NewsletterSignup.tsx]: The heading and the privacy policy link are centering, but they should be left aligned.  I think the `alignItems` prop will need to be added back to control that alignment.

File: src/components/NewsletterSignup/NewsletterSignup.mdx
Comment (bigfishdesign13) [inline src/components/NewsletterSignup/NewsletterSignup.mdx:207]: Add a code example showing the `highlightColor` prop.

File: src/components/NewsletterSignup/NewsletterSignup.mdx
Comment (bigfishdesign13) [inline src/components/NewsletterSignup/NewsletterSignup.mdx]: To be better aligned with other DS docs, change the end of this sentence to the following...
```
...a custom `Heading` element can be passed instead.
```

File: src/components/NewsletterSignup/NewsletterSignup.mdx
Comment (bigfishdesign13) [inline src/components/NewsletterSignup/NewsletterSignup.mdx]: Remove this.

File: src/components/NewsletterSignup/NewsletterSignup.stories.tsx
Comment (bigfishdesign13) [inline src/components/NewsletterSignup/NewsletterSignup.stories.tsx]: Change to `Education Newsletter Color`.

File: src/components/NewsletterSignup/NewsletterSignup.stories.tsx
Comment (bigfishdesign13) [inline src/components/NewsletterSignup/NewsletterSignup.stories.tsx]: Change to `Schomburg Newsletter Color`.

**Review by bigfishdesign13 — CHANGES_REQUESTED** [review]
Looks good.  I made some recommendations for the docs, but there is one style change that needs to be addressed before merging.

File: src/components/NewsletterSignup/NewsletterSignup.mdx
Comment (isastettler) [reply src/components/NewsletterSignup/NewsletterSignup.mdx:207]: Added.

File: src/components/NewsletterSignup/NewsletterSignup.mdx
Comment (isastettler) [reply src/components/NewsletterSignup/NewsletterSignup.mdx]: Done.

File: src/components/NewsletterSignup/NewsletterSignup.mdx
Comment (isastettler) [reply src/components/NewsletterSignup/NewsletterSignup.mdx]: Done.

File: src/components/NewsletterSignup/NewsletterSignup.stories.tsx
Comment (isastettler) [reply src/components/NewsletterSignup/NewsletterSignup.stories.tsx]: Done.

File: src/components/NewsletterSignup/NewsletterSignup.stories.tsx
Comment (isastettler) [reply src/components/NewsletterSignup/NewsletterSignup.stories.tsx]: I removed the second example because with the code snipped added I thought it makes more sense just to have one that first shows the code and then the rendered component.

File: src/components/NewsletterSignup/NewsletterSignup.tsx
Comment (isastettler) [reply src/components/NewsletterSignup/NewsletterSignup.tsx]: Oh good catch! Added it back. Removed the `alignItems` in the pitch section of the theme file as it was targeting the same section but without success.

File: src/components/NewsletterSignup/NewsletterSignup.stories.tsx
Comment (bigfishdesign13) [reply src/components/NewsletterSignup/NewsletterSignup.stories.tsx]: Ok, makes sense.

**Review by bigfishdesign13 — APPROVED** [review]
Looks good!

**Pull request merged**
